spec/lib/whitehall_importer/integrity_checker_spec.rb

```diff
@@ -90,6 +90,57 @@
       integrity_check = WhitehallImporter::IntegrityChecker.new(edition)
       expect(integrity_check.valid?).to be true
     end
+
+    context "with an attachment not yet on asset mananger" do
+      let(:organisation_service) { instance_double(Organisations) }
+      let(:file_attachment_revision) { create(:file_attachment_revision) }
+
+      let(:edition) do
+        build(
+          :edition,
+          document_type: document_type,
+          tags: {
+            primary_publishing_organisation: [SecureRandom.uuid],
+            organisations: [SecureRandom.uuid],
+          },
+          file_attachment_revisions: [file_attachment_revision],
+          contents: {
+            body: "[InlineAttachment:#{file_attachment_revision.filename}]",
+          },
+        )
+      end
+
+      let(:publishing_api_item) do
+        {
+          content_id: edition.content_id,
+          base_path: edition.base_path,
+          title: edition.title,
+          description: edition.summary,
+          document_type: edition.document_type.id,
+          schema_name: edition.document_type.publishing_metadata.schema_name,
+          details: {
+            body: GovspeakDocument.new(edition.contents["body"], edition).payload_html,
+          },
+          links: {
+            primary_publishing_organisation: edition.tags["primary_publishing_organisation"].to_a,
+            organisations: edition.tags["organisations"].to_a + edition.tags["primary_publishing_organisation"].to_a,
+          },
+        }
+      end
```

Comment:
This test looks to have a lot of setup outside of the code that we are testing. I'm guessing all we're really testing is that it doesn't blow up without an attachment. 

Do errors occur if we don't have the tags?

I wonder if we could use a method in this test to setup the Publishing API stub/item as we repeat that block of around ~10-15 lines about 3 times (and then also [modify it](https://github.com/alphagov/content-publisher/blob/9770b5d51c988ea461501827c411a64d6c2ea35d/spec/lib/whitehall_importer/integrity_checker_spec.rb#L50-L52) in other places) - I'd guess we mostly only change one or two aspects so may to pass in an edition and a hash of overrides.

Reply:
We do need the tags (there's a stubbing inconsistency if we don't).  I've moved the set up of the Publishing API response into a method of its own, which can be modified by passing in a hash.

Reply:
Great - the code has definitely benefitted from adding that method.

I just had a play myself to see if we can remove the orgs set-up and I think it's doable with a couple of tweaks: https://github.com/alphagov/content-publisher/commit/33d67067f354beba7d016adf80116692f63c8d52

Reply:
Thanks, I've taken the approach you've suggested.